src/main/java/com/mycompany/myapp/service/UserService.java

```diff
@@ -143,48 +146,49 @@ public UserService(UserRepository userRepository, AuthorityRepository authorityR
 
     /**
      * Gets a list of all the authorities.
+     *
      * @return a list of all the authorities.
      */
     public Flux<String> getAuthorities() {
         return authorityRepository.findAll().map(Authority::getName);
     }
 
-    private User syncUserWithIdP(Map<String, Object> details, User user) {
+    private Mono<User> syncUserWithIdP(Map<String, Object> details, User user) {
         // save authorities in to sync user roles/groups between IdP and JHipster's local database
-        Collection<String> dbAuthorities = getAuthorities();
-        Collection<String> userAuthorities =
-            user.getAuthorities().stream().map(Authority::getName).collect(Collectors.toList());
-        for (String authority : userAuthorities) {
-            if (!dbAuthorities.contains(authority)) {
+        Flux<String> userAuthorities = Flux.fromStream(user.getAuthorities().stream().map(Authority::getName));
+        Flux<String> newAuthorities = Flux.merge(userAuthorities, getAuthorities()).distinct();
+
+        // check to see if user's authorities exist in database, add them if they don't
+        newAuthorities
+            .filter(authority -> !authority.startsWith("SCOPE_"))
+            .flatMap(authority -> {
                 log.debug("Saving authority '{}' in local database", authority);
                 Authority authorityToSave = new Authority();
                 authorityToSave.setName(authority);
-                authorityRepository.save(authorityToSave);
-            }
-        }
+                return authorityRepository.save(authorityToSave);
+            }).subscribe();
```

Comment:
You should never have to call `subscribe` in a WebFlux project. Subscription is done internally by Spring and the methods should all be chained together to propagate the backpressure.